aces/Loris
# [Imaging Browser] Fix behaviour of Caveat link.

## Brief summary of changes

This PR accomplishes the following:

- Implements proper behaviour for the `Caveat` link that you find in the image panel associated to each scan in the imaging browser. There is a caveat drop down in this panel that allows you to set whether there is a caveat for a given scan: these are manual caveats. If set to true, the caveat label associated to this drop-down will be changed to a link that redirects to the resolved violations submenu of the MRI violations module showing the manual caveat for that scan. When not set or set to false, the label will remain as is (i.e an ordinary, non-clickable label).
- If there was a manual caveat for a given scan and the caveat drop-down is subsequently set to false, then the records in tables `mri_violations_log` and `violations_resolved` will be deleted.
- Ensures that every record in table `mri_violations_log` with column `Header` set to `Manual Caveat Set by <username>` has an associated record in table `violations_resolved`. Such a record will be created if needed in order to meet this criterium.
- Adds data in the raisin bread DB so that it meets the criterium above.

Technical note: the `Caveat` column in table `files` is set to 1 if and only if the file either has a manual caveat set by a LORIS user **or** if the MRI pipeline has created an MRI violation for that file, which in a sense is an "automatic" (i.e. non-manual) caveat. Note that a file can have both types of caveat.

#### Testing instructions (if applicable)

1. Access the imaging browser and browse the scans until you find one for which the Caveat is set to false (most of the scans on raisin bread are).
2. Change the Caveat of this scan to true and click save in the left side panel.
3. Validate that there is a `Caveat` link above the drop-down that, when clicked, takes you to the resolved violations submenu of the MRI violations module with the newly created record displayed. Note down the record's series UID.
4. Go back to the imaging browser and switch back the Caveat to false. Click `save`.
5. Make sure the link has reverted to an ordinary label.
6. Access the MRI violations module and enter the series UID of the record noted in 3 in the filter. Click `Show data`. Validate that no records with the given series UID and header set to `Manual Caveat Set by <username>` show up.


#### Link(s) to related issue(s)

* Resolves #6280
